### PR TITLE
NEW: Trigger for set status in supplier order

### PR DESCRIPTION
**Pull request by IonAgorria**
Adds missing trigger for status modification in supplier order class

### CODE REVIEW

File: htdocs/fourn/class/fournisseur.commande.class.php [hunk 1]
```diff
@@ -2104,7 +2104,10 @@ function setStatus($user,$status)
         $resql = $this->db->query($sql);
         if ($resql)
         {
-            //TODO: Add trigger for status modification
+            // Call trigger
+            $result=$this->call_trigger('ORDER_SUPPLIER_SET_STATUS',$user);
```
Comment (eldy) [inline htdocs/fourn/class/fournisseur.commande.class.php]: To be consistent with rest of code, triggers name must depends on status and use already existing value.
For exemple:
ORDER_SUPPLIER_CANCEL
ORDER_SUPPLIER_REFUSE
ORDER_SUPPLIER_APPROVE
ORDER_SUPPLIER_CLASSIFY_BILLED

**Comment by IonAgorria** [comment]
I was tempted to use "REOPEN" because this method is used when reopening, but the method itself is can set any status, so I chose SET_STATUS.

It's REOPEN instead ok?

**Comment by eldy** [comment]
If means, you must, inside the function setStatus, test the value of the new status to use the correct string depending on this new status, so use several strings (one per status) instead of one global string.

**Comment by IonAgorria** [comment]
I added the _STATUS_ between to avoid collisions with existing triggers

**Pull request merged**